test-result-summary-client/src/Dashboard/Widgets/Graph/DayTrader7.jsx

```diff
@@ -0,0 +1,249 @@
+import { Checkbox } from 'antd';
+import BenchmarkMath from '../../../PerfCompare/lib/BenchmarkMath';
+import math from 'mathjs';
+import utils from './utils';
+
+const map = {
+    "Test_openjdk8_j9_sanity.perf_x86-64_linux_Liberty": "Test_openjdk8_j9_sanity.perf_x86-64_linux_Liberty",
+};
+let display = {
+    "Test_openjdk8_j9_sanity.perf_x86-64_linux_Libertym": true,
+};
+//currently no baseline runs are made we use hard coded baseline values for scaling graph
+let baseLineValue = 300000;
+let scale = baseLineValue/100;
+let baseLineValuesu = 13000;
+let scalesu = baseLineValuesu/100;
+export class DayTrader3Setting extends Component {
+    onChange = obj => {
+        for (let i in display) {
+            display[i] = false;
+        }
+        for (let j in obj) {
+            display[obj[j]] = true;
+        }
+        this.props.onChange( { buildSelected: obj[obj.length -1] } );
+    }
+
+    render() {
+        return <div style={{ maxWidth: 400 }}>
+            <Checkbox.Group onChange={this.onChange} values={map.keys} defaultValue={["Daily-Liberty-DayTrader3"]}>
+                {Object.keys( map ).map( key => {
+                    return <Checkbox key={key} value={key} checked={false}>{map[key]}</Checkbox>;
+                } )}
+            </Checkbox.Group>
+        </div>
+    }
+}
+
+export default class DayTrader3 extends Component {
+    static Title = props => map[props.buildSelected] || '';
+    static defaultSize = { w: 2, h: 4 }
+    static Setting = DayTrader3Setting;
+    static defaultSettings = {
+        buildSelected: Object.keys( map )[0]
+    }
+
+    state = {
+        displaySeries: [],
+    };
+
+    async componentDidMount() {
+        await this.updateData();
+    }
+
+
+    async componentDidUpdate( prevProps ) {
+        if ( prevProps.buildSelected !== this.props.buildSelected ) {
+            await this.updateData();
+        }
+    }
+
+    async updateData() {
+
+        const { buildSelected } = this.props;
+        const buildName = encodeURIComponent( buildSelected );
+        const response = await fetch( `/api/getBuildHistory?type=Perf&buildName=${buildName}&status=Done&limit=100&asc`, {
+            method: 'get'
+        } );
+        const results = await response.json();
+        const startupresultsByJDKBuild = {};
+        const footprintresultsByJDKBuild = {};
+        let startupTime = [];
+        let footprint = []
+        let baseLine = [];
+        results.forEach(( t, i ) => {
+            if (t.buildResult == "SUCCESS" || t.validAggregateInfo) {
+            for ( let element of t.aggregateInfo ) {
+                const jdkDate = t.timestamp;
+                if(!footprintresultsByJDKBuild[t.buildName]) {
+                    footprintresultsByJDKBuild[t.buildName] = {};
+                }
+                footprintresultsByJDKBuild[t.buildName][jdkDate] = footprintresultsByJDKBuild[jdkDate] || [];
+                footprintresultsByJDKBuild[t.buildName][jdkDate].push( {
```

Comment:
Why we need to add `element.metrics[0]` and `element.metrics[1]` specifically? Can we use a loop here?